src/SourceBuild/tarball/content/patches/roslyn-analyzers/0002-Update-versions.props-source-build-behavior.patch

```diff
@@ -0,0 +1,63 @@
+From c989fd8d4e2ab9dc6d0a94bdaf43a2f5ec0a9d45 Mon Sep 17 00:00:00 2001
+From: Michael Simons <msimons@microsoft.com>
+Date: Wed, 11 Aug 2021 22:12:31 +0000
+Subject: [PATCH] Update versions.props source build behavior
+
+These changes allow the repo to more easily control which dependency versions are used during source-build.
+Normally the source build infrastructure imports its versions.props file containing all the source built versions after
+the repo's versions.props file is imported.
+---
+ eng/Versions.props | 28 ++++++++++++++++------------
+ 1 file changed, 16 insertions(+), 12 deletions(-)
+
+diff --git a/eng/Versions.props b/eng/Versions.props
+index b0c147944..c41b54826 100644
+--- a/eng/Versions.props
++++ b/eng/Versions.props
+@@ -1,4 +1,20 @@
+ <Project>
++  <Import Project="$(SourceBuildPackageVersionPropsPath)" Condition="'$(SourceBuildPackageVersionPropsPath)' != ''"/>
++
++  <PropertyGroup>
++    <!--
++      Binaries that need to be executable during source build are restricted to versions available within source build.
++      This section defines executable versions of packages which are referenced via lower-version reference assemblies
++      during the build.
++    -->
++    <SystemCollectionsImmutableExecutableVersion>5.0.0</SystemCollectionsImmutableExecutableVersion>
++    <SystemCollectionsImmutableExecutableVersion Condition="$(SystemCollectionsImmutableVersion) != ''">$(SystemCollectionsImmutableVersion)</SystemCollectionsImmutableExecutableVersion>
```

Comment:
These two lines should be reversed, and the fallback to 5.0.0 should only occur if the value wasn't obtained from source build.

```suggestion
+    <SystemCollectionsImmutableExecutableVersion>$(SystemCollectionsImmutableVersion)</SystemCollectionsImmutableExecutableVersion>
+    <SystemCollectionsImmutableExecutableVersion Condition="'$(SystemCollectionsImmutableExecutableVersion)' != ''">5.0.0</SystemCollectionsImmutableExecutableVersion>
```

The same pattern can repeat for each of the executable dependencies.

Reply:
Is this what you are suggestion?

```
    <SystemCollectionsImmutableExecutableVersion>$(SystemCollectionsImmutableVersion)</SystemCollectionsImmutableExecutableVersion>
    <SystemCollectionsImmutableExecutableVersion Condition="$(SystemCollectionsImmutableExecutableVersion) != ''">5.0.0</SystemCollectionsImmutableExecutableVersion>
```

They appear functionally equivalent.

Reply:
Yes, they are functionally equivalent. However, the form I gave will avoid some messages in the binlog about property reassignment.

Reply:
Note that the `Condition` in https://github.com/dotnet/installer/pull/11114#discussion_r691271463 is missing a pair of single-quotes.